### PR TITLE
Fix inconsistency in slash command option naming

### PR DESCRIPTION
**Pull request by Vampire**

### CODE REVIEW

File: javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteractionOption.java [hunk 1]
```diff
@@ -135,7 +135,7 @@ default boolean isSubcommandOrGroup() {
      *
      * @return The mentionable value of this option.
      */
```
Comment (KILLEliteMaste) [inline javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteractionOption.java:137]: Can't select it all but the Javadoc is wrong

File: javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteractionOptionsProvider.java [hunk 2]
```diff
@@ -168,11 +168,11 @@
      *
      * @return An Optional with the double value of such an option if it exists; an empty Optional otherwise
      * @deprecated Use {@link #getOptions() getOptions().get(0).getNumberValue()}
-     *     or {@link #getOptionNumberValueByIndex(int) getOptionNumberValueByIndex(0)} instead.
+     *     or {@link #getOptionDecimalValueByIndex(int) getOptionDecimalValueByIndex(0)} instead.
      */
     @Deprecated
     default Optional<Double> getFirstOptionNumberValue() {
```
Comment (KILLEliteMaste) [inline javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteractionOptionsProvider.java:174]: Won't mark every place but the same should be done for `Second` and `Third` NumberValue
```suggestion
    default Optional<Double> getFirstOptionDecimalValue() {
```

File: javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteractionOptionsProvider.java [hunk 3]
```diff
@@ -168,11 +168,11 @@
      *
      * @return An Optional with the double value of such an option if it exists; an empty Optional otherwise
      * @deprecated Use {@link #getOptions() getOptions().get(0).getNumberValue()}
```
Comment (KILLEliteMaste) [inline javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteractionOptionsProvider.java:170]: Won't mark every place but the same should be done for `Second` and `Third` NumberValue
```suggestion
     * @deprecated Use {@link #getOptions() getOptions().get(0).getDecimalValue()}
```

File: javacord-api/src/main/java/org/javacord/api/interaction/internal/SlashCommandOptionBuilderDelegate.java [hunk 4]
```diff
@@ -86,32 +86,32 @@
...
      */
-    void setIntegerMaxValue(long integerMaxValue);
+    void setLongMaxValue(long longMaxValue);
 
     /**
-     * Sets the {@link SlashCommandOptionType#NUMBER} min value for the slash command option.
+     * Sets the {@link SlashCommandOptionType#DECIMAL} min value for the slash command option.
      *
-     * @param numberMinValue The integer min value.
+     * @param decimalMinValue The number min value.
```
Comment (KILLEliteMaste) [inline javacord-api/src/main/java/org/javacord/api/interaction/internal/SlashCommandOptionBuilderDelegate.java:105]:
```suggestion
     * @param decimalMinValue The decimal min value.
```

File: javacord-api/src/main/java/org/javacord/api/interaction/internal/SlashCommandOptionBuilderDelegate.java [hunk 5]
```diff
@@ -86,32 +86,32 @@
...
      */
-    void setNumberMinValue(double numberMinValue);
+    void setDecimalMinValue(double decimalMinValue);
 
     /**
-     * Sets the {@link SlashCommandOptionType#NUMBER} max value for the slash command option.
+     * Sets the {@link SlashCommandOptionType#DECIMAL} max value for the slash command option.
      *
-     * @param numberMaxValue The integer max value.
+     * @param decimalMaxValue The number max value.
```
Comment (KILLEliteMaste) [inline javacord-api/src/main/java/org/javacord/api/interaction/internal/SlashCommandOptionBuilderDelegate.java:112]:
```suggestion
     * @param decimalMaxValue The decimal max value.
```

File: javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandOptionType.java [hunk 6]
```diff
@@ -18,7 +18,7 @@
     /**
      * Any double between -2^53 and 2^53.
```
Comment (KILLEliteMaste) [inline javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandOptionType.java:19]:
```suggestion
     * Any double between -2^53 and 2^53. This is the NUMBER option according to the Discord docs.
```

File: javacord-core/src/main/java/org/javacord/core/interaction/SlashCommandInteractionOptionImpl.java [hunk 7]
```diff
@@ -35,7 +35,7 @@
     private final Long channelValue;
     private final Long roleValue;
     private final Long mentionableValue;
-    private final Double numberValue;
+    private final Double decimalValue;
```
Comment (KILLEliteMaste) [inline javacord-core/src/main/java/org/javacord/core/interaction/SlashCommandInteractionOptionImpl.java:38]:
```suggestion
    /**
    * This is the NUMBER option according to the Discord docs.
    */
    private final Double decimalValue;
```

**Review by KILLEliteMaste — CHANGES_REQUESTED** [review]

File: javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteractionOptionsProvider.java (hunk 2 above)
Comment (Vampire) [reply javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteractionOptionsProvider.java:174]: I intentionally left those out, as they are deprecated anyway.
Renaming them now would immediately remove the old ones and introduce new methods that are right-away deprecated.
That felt strange to me, so I kept them with the old name.
What do you think?

Comment (KILLEliteMaste) [reply javacord-api/src/main/java/org/javacord/api/interaction/SlashCommandInteractionOptionsProvider.java:174]: Right, that would be a bit strange. But that applies for both options, having inconsistent naming again or like you said introducing new methods that are right away deprecated. But inconsistent naming might be a better than a breaking change for this case, so leave them as they are

**Review by KILLEliteMaste — APPROVED** [review]
LGTM!

**Pull request merged**